### PR TITLE
Bump protocol version

### PR DESCRIPTION
**Pull request by agluszak**
Should also fix the failing CI

### CODE REVIEW

File: website/generated/docs/bindings/java.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Add the following snippet to your build to add dependency on `bsp4j`.
 ### Gradle
 
 ```groovy
-compile group: 'ch.epfl.scala', name: 'bsp4j', version: '2.1.0-M6'
+compile group: 'ch.epfl.scala', name: 'bsp4j', version: '2.2.0-M1.alpha'
```
Comment (adpi2) [inline website/generated/docs/bindings/java.md:19]: Where does this `-M1.alpha` comes from?

File: website/generated/docs/extensions/cargo.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ protocol.
 
 ## BSP version
 
-`2.1.0`
+`2.2.0`
```
Comment (adpi2) [inline website/generated/docs/extensions/cargo.md:12]: When will this change be published to the website?

**Review by adpi2 — COMMENTED** [review]
A few questions to compensate for my ignorance

**Comment by agluszak** [comment]
@adpi2 I made a pre-release a few days ago with the newest changes: https://github.com/build-server-protocol/build-server-protocol/releases/tag/v2.2.0-M1.alpha

File: website/generated/docs/bindings/java.md (hunk 1 above)
Comment (agluszak) [reply website/generated/docs/bindings/java.md:19]: from the newest tag on git

**Comment by agluszak** [comment]
@adpi2 can you accept the PR? or do you have further questions?

**Comment by jastice** [comment]
I don't think we need `.alpha` on top of `M1`, which is already meant as intermediate / pre-release

**Comment by agluszak** [comment]
M1 is meant as intermediate/pre-release? It looks like so far we've been using versioning scheme `{protocol_version}-M{library_version}`

**Comment by jastice** [comment]
> M1 is meant as intermediate/pre-release? It looks like so far we've been using versioning scheme {protocol_version}-M{library_version}

No, `M-x` was always just an incremented number before a "proper" release.

**Comment by lefou** [comment]
`2.1.0-M5` means "Milestone 5" toward a `2.1.0` release.

**Comment by agluszak** [comment]
Okay, so maybe let's discuss the versioning scheme we want to use from now. It looks like we have never had a proper 2.1.0 release then. The new additions to the protocol should land in 2.2.0 IMO. And we had 2.1.0-M6 release which didn't contain any breaking changes to the protocol, but broke some stuff in the libraries because of being auto-generated.

**Comment by lefou** [comment]
We could tag `2.1.0-M5` also as `2.1.0` making the version which is currently used/implemented by most tools a stable one. 

After that, we could start to split the versions.

Next BSP version is `2.2.0`, if it's not breaking with `2.1.0`, otherwise `3.0.0`.

For libraries, we take the classic semantic versioning approach, based on the ABI. Each library has a programmatic way to provide the implemented BSP version. Question is, do we need to encode the implemented version in the library name? I think not, as long as the chanelog states which versions match. Or we provide a version matrix in the documentation.

**Review by jastice — APPROVED** [review]

**Pull request merged**